src/topology/builder.rs

```diff
@@ -27,16 +28,18 @@ use tokio::{
     select,
     time::{timeout, Duration},
 };
-use vector_core::ByteSizeOf;
 use vector_core::{
-    buffers::{BufferInputCloner, BufferType},
+    buffers::{BufferInputCloner, BufferStream, BufferType},
     internal_event::EventsSent,
+    ByteSizeOf,
 };
 
 lazy_static! {
     static ref ENRICHMENT_TABLES: enrichment::TableRegistry = enrichment::TableRegistry::default();
 }
 
+static TRANSFORM_CONCURRENCY_LIMIT: Lazy<usize> = Lazy::new(num_cpus::get);
```

Comment:
Should this instead default to the number of available threads for Tokio? That also defaults to `num_cpu` but users can configure it via `VECTOR_THREADS`.